pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -246,6 +246,27 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(ctx context.Context,
 		msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v",
 			instance.Spec.VMName, err)
 		log.Error(msg)
+		// If virtualmachine instance is NotFound and if deletion timestamp is set on CnsFileAccessConfig instance,
+		// then proceed with the deletion of CnsFileAccessConfig instance.
+		if apierrors.IsNotFound(err) && instance.DeletionTimestamp != nil {
+			log.Infof("CnsFileAccessConfig instance %q has deletion timestamp set, but VM instance with "+
+				"name %q is not found. Processing the deletion of CnsFileAccessConfig instance.",
+				instance.Name, instance.Spec.VMName)
+			removeFinalizerFromCRDInstance(ctx, instance)
+			err = updateCnsFileAccessConfig(ctx, r.client, instance)
+			if err != nil {
+				msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+					instance.Name, instance.Namespace, err)
+				recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+			// Cleanup instance entry from backOffDuration map.
+			backOffDurationMapMutex.Lock()
+			delete(backOffDuration, instance.Name)
+			backOffDurationMapMutex.Unlock()
+			return reconcile.Result{}, nil
```

Comment:
If we return success here, we will miss the deletePermission on the fileshare.

I think we need to invoke 
"err = r.configureNetPermissionsForFileVolume(ctx, volumeID, vm, instance, true)".

I'm not sure how you can fetch the vm info though.

Reply:
Yes, since VM is already deleted we won't get VM object. So, it is not possible to run configureNerPermissionsForFileVolume. I am also not sure if we can do anything else here.
@divyenpatel Can you please review these changes once.

Reply:
Can you check if we can get VM IP address from cnsfilevolumeclients.cns.vmware.com ?

if we can get IP from this CR instance then we can use that IP and remove ACL.

Also we need to add a check if PVC for this is already deleted then there is no need to call API to remove ACL and we can remove finalizer.

Reply:
Updated the code to fetch VM IP from cnsfilevolumeclients.cns.vmware.com